doc/usage/configuration.rst

```diff
@@ -808,6 +808,25 @@ General configuration
 
    .. versionadded:: 5.1
 
+.. confval:: copyright_build_year_substitution
```

Comment:
I'm not sure that an additional configuration variable has benefits--setting the environment variable is a key signal in and of itself -- some things are sacrificed in pursuit of reproducibility.

Reply:
Mm.  It's a proxy for "I attest that I've configured static copyright notices, so I don't want/need substitution to occur even during reproducible builds".

We do have an example of a project, `matplotlib`, that has intentionally chosen to use a copyright notice format that isn't matched by the substitution logic -- that's only one datapoint, but I do think that other projects might want to avoid their copyright notices being adjusted automatically, even against the potential upside of bit-for-bit builds.

Reply:
Also note: if we could detect that the copyright field(s) (must remember about `epub_copyright` also) _are_ definitely configured statically in `conf.py`, that would remove the need for the setting.  But that'd require parsing (not simply importing) the `conf.py` file during the build.

Reply:
I've removed the `copyright_build_year_substitution` config setting from the changes in this branch, and opened draft #12519 to add detection of static (constant) copyright notice configuration using Python `ast` parsing at config load-time.